hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/ComplexAvroKeyGenerator.java

```diff
@@ -32,7 +33,8 @@ public class ComplexAvroKeyGenerator extends BaseKeyGenerator {
 
   public ComplexAvroKeyGenerator(TypedProperties props) {
     super(props);
-    this.recordKeyFields = Arrays.stream(props.getString(KeyGeneratorOptions.RECORDKEY_FIELD_NAME.key()).split(","))
+    this.recordKeyFields = autoGenerateRecordKeys() ? Collections.emptyList() :
```

Comment:
yes, unfortunately, the way we parse record keys is specific to each key gen class and is not generalized in any base class. Even then, the base class needs some fixes since record key was deemed mandatory before and now w/ this patch, it is not. 
I will work on a follow up patch after landing this (do not want to drag this further) to fix the parsing or record keys. 
https://issues.apache.org/jira/browse/HUDI-6172